src-lib/HTMell/Content.hs

```diff
@@ -0,0 +1,14 @@
+-- | HTMell content related types and functions
+module HTMell.Content
+    ( HTMellContent(..)
+    ) where
+
+import System.FilePath ( FilePath )
+import Data.Map ( Map )
+import Data.Text ( Text )
+
+-- | Some content stored in 'HTMell.Tree.HNode' trees
+class HTMellContent c where
+    getContent :: FilePath -> Maybe c
+    metadata :: c -> Map String String
+    html :: c -> Text
```

Comment:
These methods (`getContent`, `metadata`, `html`) need to be documented with Haddock!

Reply:
Done in an updated version of that commit: 06da8c4